src/admin_extra/service_admin_extra_roster.erl

```diff
@@ -175,20 +179,35 @@ add_rosteritem(LocalUser, LocalServer, User, Server, Nick, Group, Subs) ->
     end.
 
 %% @doc returns result of mnesia or rdbms transaction
--spec subscribe(LocalUser :: jid:user(),
+%% add to roster directly, without generating a roster push (in case we do bulk inserts)
+-spec subscribe_silent(LocalUser :: jid:user(),
```

Comment:
Looks like this is something new we didn't have before. What's the reason behind this new function and where exactly is it used?

Reply:
`mongooseimctl` (more specifically `service_admin_extra_roster`) adds roster entries in two ways: either one by one or in bulk. When it adds one by one tests expect roster push to be sent; for bulk inserts push should not be sent. Previously, `service_admin_extra_roster` used `mod_roster:set_items` for all operations, and for single inserts it sent iq push itself. Since now it is `mod_roster` that does the pushing, service_admin_extra_roster has to call the right api (`mod_roster:set_roster_entry` if it wants the push to be sent, or `mod_roster:set_items` if it wants it silent. Hence the split into `subscribe` and `subscribe_silent`.

Reply:
All right, thanks for the explanation!